### PR TITLE
Use command line options to determine what required libraries to check for

### LINKED ISSUE #576: CEKit fails to run when "docker-squash" is missing, even when disabled
**Describe the bug**

When "docker-squash" is missing, then cekit fails even when the argument `--no-squash` is being used.

**To reproduce**

Run cekit build on a machine where docker squash is missing, eg. RHEL.

**Expected behavior**

Failing is ok, when `--no-squash` is omitted, but when it is present, it should not fail.

**Logs**

~~~
cekit --descriptor image.yaml build  --overrides "{'name': 'amq7-tech-preview/amq-online-1-iot-device-registry-datagrid'}" docker --no-squash  
2019-07-12 16:06:05,691 cekit        INFO     You are running on known platform: Red Hat Enterprise Linux Workstation 7.6 (Maipo)
2019-07-12 16:06:05,789 cekit        INFO     Generating files for docker engine
2019-07-12 16:06:05,794 cekit        INFO     Initializing image descriptor...
2019-07-12 16:06:05,907 cekit        INFO     Preparing resource 'cct_module'
2019-07-12 16:06:08,346 cekit        INFO     Handling artifacts...
2019-07-12 16:06:08,346 cekit        INFO     Preparing resource 'xx.jar'
2019-07-12 16:06:08,467 cekit        INFO     Using cached artifact 'xx.jar'.
2019-07-12 16:06:08,467 cekit        WARNING  The image has ContentSets repositories specified, all other repositories are removed!
2019-07-12 16:06:08,471 cekit        INFO     Using Red Hat ODCS service to create composes
2019-07-12 16:06:14,444 cekit        INFO     Preparing resource 'odcs-141137.repo'
2019-07-12 16:06:14,757 cekit        INFO     Rendering Dockerfile...
2019-07-12 16:06:14,809 cekit        ERROR    Required CEKit library 'docker-squash' was not found; required module 'docker_squash' could not be found.
~~~

**Environment information:**
 - Operating system:  Red Hat Enterprise Linux Workstation 7.6 (Maipo)
 - CEKit version: Result of `cekit --version`: 3.1.0

### PR DESCRIPTION
**Pull request by rnc**
Fixes #576

### CODE REVIEW

File: cekit/cli.py [hunk 1]
```diff
@@ -236,8 +236,7 @@ def top_context(ctx):
 def prepare_params(ctx):
     main_context = top_context(ctx)
     common_params = Map(main_context.params)
-
-    params = Map({})
+    params = Map(main_context.params)
```
Comment (goldmann) [inline cekit/cli.py]: This should be taken even further. We should reuse logic similar to the one in `top_context` method, a recursive way of combining all params. It should prepare a dict with all params and we should use it directly here.

File: cekit/tools.py [hunk 2]
```diff
@@ -377,3 +377,18 @@ def handle(self, o, params):
                 # Execute that method to get list of dependencies and try to handle them
                 self._handle_dependencies(o.dependencies(params))
                 return
+
+
+def merge_two_dicts(x, y):
```
[the remark this answers is not in the document]
Comment (rnc) [reply cekit/tools.py]: Only used in tests at the moment.

[the remark this answers is not in the document]
Comment (goldmann) [reply cekit/tools.py]: If it's required I would create a helper method in the test file.

File: cekit/cli.py (hunk 1 above)
Comment (rnc) [reply cekit/cli.py]: Potentially. I'd done minimal changes to cli until a second pass to deal with the config object was done. Can discuss tomorrow.

File: cekit/tools.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (rnc) [reply cekit/tools.py]: Yeah some form of common test utils makes sense I guess

File: cekit/cli.py (hunk 1 above)
Comment (goldmann) [reply cekit/cli.py]: I was thinking of removing the `top_context` method and rewrite the `prepare_params` in something like (warning, not validated/checked):

```python
def prepare_params(ctx, params=None):
    if params is None:
        params = {}

    if ctx.parent:
        prepare_params(ctx.parent, params)

    params.update(ctx.params)

    return params
```

Comment (goldmann) [reply cekit/cli.py]: Additionally conversion to `Map` object should be performed.

File: cekit/tools.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (rnc) [reply cekit/tools.py]: Done

File: cekit/cli.py [hunk 3]
```diff
@@ -108,7 +108,7 @@ def build_docker(ctx, pull, no_squash, tags):  # pylint: disable=unused-argument
 
 
 @build.command(name="buildah", short_help="Build using Buildah engine")
-@click.option('--pull', help="Always try to fetch latest base image.", is_flag=True)
+@click.option('--pull', help="Always try to fetch latest'dict' object has no attribute 'verbose' base image.", is_flag=True)
```
Comment (goldmann) [inline cekit/cli.py]: Ooops

**Comment by rnc** [comment]
retest this please

**Comment by goldmann** [comment]
I guess you need to rebase and force-push to kick new test run.

**Comment by goldmann** [comment]
Or, hold on.

**Comment by goldmann** [comment]
Kicked it.

**Comment by goldmann** [comment]
Thank you!

**Pull request merged**